### DISCUSSION TITLE
Conditional Assignment shorthand, for Left Side

### DISCUSSION THREAD

**Discussion by najak3d** [dotnet/csharplang#5831] [Language Ideas] [answered]
I couldn't find a notation that is equivalent to the ternary '?' operator for right-side, such as:

     float result = (boolExpression) ? val1 : val2;

What would be very nice is if there is a short-hand for selecting the Left-side variable that will received the assignment, something like this:

```c#
  float result1, result2;
     
  ((boolExpression) ? result1 : result2) = result;
```

This avoids the common multi-line syntax such as:

```c#
  float result1, result2;
  
  if (boolExpression)
      result1 = result;
  else
      result2 = result;
```

I'm unaware of a shorthand alternative that is equivalent to this longhand notation.

**Comment by HaloFour** [comment]
How common is it that you need to assign the result of an expression to exactly one of two different targets based on some conditional?

**Reply by najak3d** [reply]
It happened to me just now inside of a single method, and occurred to me that this was something I had wondered dozens of times before.   And happened yesterday as well, but forgot the location.

Here's the code I'm working on now, that could have used short-hand:

```c#
	if (tupleIndex == 1)
		def.min = double.Parse(item);
	else
		def.max = double.Parse(item);
```

I'd have preferred to just write:
```c#
     ((tupleIndex == 1) ? def.min : def.max) = double.Parse(item);
```
This seems intuitive to me, and would work the same as it does for the right-side.

===
I'm not sure how frequently I run into this.   Often enough that it became a repeated thought, which it why I decided to raise the issue here, and see if:

1. Maybe there is some existing short-hand that I'm not aware of.
2. OR... see if this is something others would find useful.
3. OR... I'm just here to fulfill my mission of making silly proposals for others to shoot down. :)

I don't feel strongly about needing this syntax; it just seemed to me like "missing syntax" for the left-side.

**You do raise a vital point that it's probably RARE, and not worth even trying to raise this idea to become a "proposal".**

**Comment by huoyaoyuan** [comment, marked as answer]
If the LHS is something you can take address (local variable, array element, or field instead of property), you can use ref expression for it.

```csharp
int x = 0, y = 0;
(a ? ref x : ref y) = 1;
```

**Reply by najak3d** [reply]
Thank you for the suggestion; that is interesting.  It's probably a little too weird to use as a convention.    If we could remove the "ref" it would be better.

I tried your suggestion and confirmed that the following works just fine:

  ((tupleIndex == 1) ? ref def.min : ref def.max) = double.Parse(item);
